### PR TITLE
James 2213: JMAP support isForwarded flag

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/OldKeyword.java [hunk 1]
```diff
@@ -30,20 +30,24 @@
     private final Optional<Boolean> isFlagged;
     private final Optional<Boolean> isAnswered;
     private final Optional<Boolean> isDraft;
+    private final Optional<Boolean> isForwarded;
 
     @VisibleForTesting
-    OldKeyword(boolean isUnread, boolean isFlagged, boolean isAnswered, boolean isDraft) {
```
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/OldKeyword.java]: We may think of introducing a builder for this class.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 2]
```diff
@@ -396,7 +396,55 @@ public void getMessageListUnsetAnsweredFilterShouldReturnOnlyNotAnsweredMessages
...
-    
+
+    @Test
+    public void getMessageListSetForwardedFilterShouldReturnOnlyForwardedMessages() throws Exception {
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, alice, "mailbox");
+
+        ComposedMessageId messageNotForwarded = mailboxProbe.appendMessage(alice, MailboxPath.forUser(alice, "mailbox"),
+            new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+        ComposedMessageId messageForwarded = mailboxProbe.appendMessage(alice, MailboxPath.forUser(alice, "mailbox"),
+            ClassLoader.getSystemResourceAsStream("eml/twoAttachments.eml"), new Date(), false, FlagsBuilder.builder().add("Forwarded").build());
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: new Flags("Forwarded")

Comment (chibenwa) [reply server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: BTW isn't it `$Fowarded` ?

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/FilterToSearchQuery.java [hunk 3]
```diff
@@ -76,6 +76,7 @@ private SearchQuery convertCondition(FilterCondition filter) {
         filter.getIsDraft().ifPresent(isDraft -> searchQuery.andCriteria(SearchQuery.flag(Flag.DRAFT, isDraft)));
         filter.getIsFlagged().ifPresent(isFlagged -> searchQuery.andCriteria(SearchQuery.flag(Flag.FLAGGED, isFlagged)));
         filter.getIsUnread().ifPresent(isUnread -> searchQuery.andCriteria(SearchQuery.flag(Flag.SEEN, !isUnread)));
+        filter.getIsForwarded().ifPresent(isForwarded -> searchQuery.andCriteria(SearchQuery.flagSet("Forwarded", isForwarded)));
```
Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/utils/FilterToSearchQuery.java]: Could you use the constant String value introduced in `Keyword.java` ?

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/OldKeyword.java (hunk 1 above)
Comment (rouazana) [reply server/protocols/jmap/src/main/java/org/apache/james/jmap/model/OldKeyword.java]: not needed IMHO: we won't introduce other keyworkds, and this class is meant to be removed (soon I hope)

**Comment by rouazana** [comment]
> Can we get a GetMessages integration test showing isFowarded property on every messages?

The test is already here \o/
https://github.com/linagora/james-project/blob/master/server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMessages.feature#L399

**Comment by chibenwa** [comment]
No it's not!

You never check about the isForwarded property. The goal of that test would be to check the associated is property, introduced in that PR, not the `$Forwarded` keyword.

**Comment by rouazana** [comment]
> No it's not!

Ok sorry, done.

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java [hunk 4]
```diff
@@ -396,7 +396,55 @@ public void getMessageListUnsetAnsweredFilterShouldReturnOnlyNotAnsweredMessages
                     containsInAnyOrder(messageNotAnswered.getMessageId().serialize()), 
                     not(containsInAnyOrder(messageAnswered.getMessageId().serialize()))));
     }
-    
+
+    @Test
+    public void getMessageListSetForwardedFilterShouldReturnOnlyForwardedMessages() throws Exception {
```
Comment (mbaechler) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java]: s/Set/With ?

**Pull request merged**